src/main/java/com/google/sps/servlets/ChatServlet.java

```diff
@@ -0,0 +1,65 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity; 
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableList.Builder;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Streams;
+import com.google.template.soy.SoyFileSet;
+import com.google.template.soy.tofu.SoyTofu;
+import java.io.File;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/chat")
+public class ChatServlet extends HttpServlet{
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+    // Calls query on all entities of type Message
+    Query messageQuery = new Query("Message").addSort("timestamp", SortDirection.ASCENDING);
+    PreparedQuery preparedMessageQuery = datastore.prepare(messageQuery);
+
+    // Creates list of the messages text
+    ImmutableList<String> messagesList = Streams.stream(preparedMessageQuery.asIterable()).map(message -> 
+        (String) message.getProperty("message-text")).collect(ImmutableList.toImmutableList());
```

Comment:
Nit: you can import the static function separately. `import com.google.common.collect.ImmutableList.toImmutableList;`